src/main/java/stream/flarebot/flarebot/commands/moderation/PurgeCommand.java

```diff
@@ -101,6 +103,80 @@ public void run() {
         } else {
             MessageUtils.sendUsage(this, channel, sender);
         }
+    }*/
+
+    @Override
+    public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Message message, String[] args, Member member) {
+        if(args.length >= 2) {
+            // clean all 5
+            User targetUser = null;
+            if(!args[0].equalsIgnoreCase("all")) {
+                targetUser = GeneralUtils.getUser(args[0], guild.getGuildId(), true);
+                if(targetUser == null) {
+                    MessageUtils.sendErrorMessage("That target user cannot be found, try mentioning them, using the user ID or using `all` to clear the entire chat.", channel);
+                    return;
+                }                
+            }
+
+            int amount = GeneralUtils.getInt(args[1], -1);
+
+            // 2 messages min
+            if(amount < 1) {
+                MessageUtils.sendErrorMessage("You must purge at least 1 message, please give a vaid purge amount.", channel);
+                return;
+            }
+
+            // This will be a successful delete so limit here.
+            if (!guild.getPermissions().isCreator(sender)) {
```

Comment:
We should really make a better system to do cooldowns globally

Reply:
Probably but right now it isn't too needed.

Reply:
Agreed